File: Java/SauceDemo/src/main/java/pages/CheckoutStepOnePage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class CheckoutStepOnePage extends Sidebar{

    By first_name_box = By.id("first-name");
    By last_name_box = By.id("last-name");
    By postal_code_box = By.id("postal-code");
    By continue_btn = By.id("continue");


    public CheckoutStepOnePage(WebDriver driver) {
        super(driver);
    }

    public void setTextInFirstNameBox(String text) {
        driver.findElement(first_name_box).sendKeys(text);
    }

    public void setTextInLastNameBox(String text) {
        driver.findElement(last_name_box).sendKeys(text);
    }

    public void setTextInPostalCodeBox(String text) {
        driver.findElement(postal_code_box).sendKeys(text);
    }

    public void clickContinueButton() {
        driver.findElement(continue_btn).click();
    }
}

File: Python/src/pages/check_out_step_one_page.py
"""
This module contains the elements of the checkout step one page
"""

from selenium.webdriver.common.by import By
from src.pages.side_bar import Sidebar

class CheckoutStepOnePage(Sidebar):
    """
    This class contains the elements of the checkout step one page
    """
    def __init__(self, driver):
        super().__init__(driver)

        # Locators
        self.first_name_box = By.ID, "first-name"
        self.last_name_box = By.ID, "last-name"
        self.postal_code_box = By.ID, "postal-code"
        self.continue_btn = By.ID, "continue"

    def set_text_in_first_name_box(self, text):
        self.driver.find_element(*self.first_name_box).send_keys(text)

    def set_text_in_last_name_box(self, text):
        self.driver.find_element(*self.last_name_box).send_keys(text)

    def set_text_in_postal_code_box(self, text):
        self.driver.find_element(*self.postal_code_box).send_keys(text)

    def click_continue_button(self):
        self.driver.find_element(*self.continue_btn).click()
